examples/kilo.cpp

```diff
@@ -794,79 +796,72 @@ bool editorProcessKeypress(const Terminal &term) {
-
-    case CTRL_KEY('q'):
-      if (E.dirty && quit_times > 0) {
+  if (c == Key::ENTER)
+    editorInsertNewline();
+  else if (c == ctrl_key('q')) {
+    if (E.dirty && quit_times > 0) {
         editorSetStatusMessage("WARNING!!! File has unsaved changes. "
           "Press Ctrl-Q %d more times to quit.", quit_times);
         quit_times--;
         return true;
       }
       return false;
-      break;
-
-    case CTRL_KEY('s'):
-      editorSave(term);
-      break;
-
-    case Key::HOME:
-      E.cx = 0;
-      break;
-
-    case Key::END:
-      if (E.cy < E.numrows)
+  }
+  else if (c == ctrl_key('q')) {
+    if (E.dirty && quit_times > 0) {
+        editorSetStatusMessage("WARNING!!! File has unsaved changes. "
+          "Press Ctrl-Q %d more times to quit.", quit_times);
+        quit_times--;
+        return true;
+      }
+      return false;
+  }
+  else if (c == ctrl_key('s')) {
+    editorSave(term);
+  }
+  else if (c == Key::HOME) {
+    E.cx = 0;
+  }
+  else if (c == Key::END) {
+    if (E.cy < E.numrows) {
         E.cx = E.row[E.cy].size;
-      break;
-
-    case CTRL_KEY('f'):
-      editorFind(term);
-      break;
-
-    case Key::BACKSPACE:
-    case CTRL_KEY('h'):
-    case Key::DEL:
-      if (c == Key::DEL) editorMoveCursor(Key::ARROW_RIGHT);
+    }
+  }
+  else if (c == ctrl_key('f')) {
+    editorFind(term);
+  }
+  else if (c == Key::BACKSPACE ||
+           c == ctrl_key('h') ||
+           c == Key::DEL) {
+    if (c == Key::DEL) editorMoveCursor(Key::ARROW_RIGHT);
       editorDelChar();
-      break;
-
-    case Key::PAGE_UP:
-    case Key::PAGE_DOWN:
-      {
-        if (c == Key::PAGE_UP) {
-          E.cy = E.rowoff;
-        } else if (c == Key::PAGE_DOWN) {
-          E.cy = E.rowoff + E.screenrows - 1;
-          if (E.cy > E.numrows) E.cy = E.numrows;
-        }
-
-        int times = E.screenrows;
-        while (times--)
-          editorMoveCursor(c == Key::PAGE_UP ? Key::ARROW_UP : Key::ARROW_DOWN);
-      }
-      break;
-
-    case Key::ARROW_UP:
-    case Key::ARROW_DOWN:
-    case Key::ARROW_LEFT:
-    case Key::ARROW_RIGHT:
-      editorMoveCursor(c);
-      break;
-
-    case CTRL_KEY('l'):
-    case Key::ESC:
-      break;
-
-    case Key::TAB:
-      editorInsertChar('\t');
-      break;
-
-    default:
+  }
+  else if (c == Key::PAGE_UP ||
+           c == Key::PAGE_DOWN) {
+    if (c == Key::PAGE_UP) {
+      E.cy = E.rowoff;
+    } else if (c == Key::PAGE_DOWN) {
+      E.cy = E.rowoff + E.screenrows - 1;
+      if (E.cy > E.numrows) E.cy = E.numrows;
+    }
+    int times = E.screenrows;
+    while (times--) {
+    editorMoveCursor(c == Key::PAGE_UP ? Key::ARROW_UP : Key::ARROW_DOWN);
+    }
+  }
+  else if (c == Key::ARROW_UP ||
+      c == Key::ARROW_DOWN ||
+      c == Key::ARROW_LEFT ||
+      c == Key::ARROW_RIGHT) {
+    editorMoveCursor(c);
+  }
+  else if (c == Key::TAB)
+    editorInsertChar('\t');
+  else
+  {
+    if (c != ctrl_key('l') ||
+           c != Key::ESC)
```

Comment:
Would you mind only changing `CTRL_KEY` to `ctrl_key` (the same with `alt`) in this PR? This change of changing the switch to if statements is unrelated and I don't think it actually makes it more readable. And also, the `kilo.cpp` file was written by somebody else and I have it here as an example how to port things to use `cpp-terminal`, so I would prefer to keep our modifications to it minimal.

Reply:
Yeah, sorry about that change. The problem is, that after changing the macros to inline functions, the code won't compile, because the compiler is complaining about ctrl_key() not beeing static. making ctrl_key() static itself causes other prbolems. If you don't have an ace I guess there only using a switch with macros, or if's with an function.